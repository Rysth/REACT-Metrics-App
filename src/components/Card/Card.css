.card {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: flex-end;
  position: relative;
  height: 11em;
  padding: 1em;
  text-decoration: none;
}

.card:active {
  background-color: var(--CL-neutral-black);
}

.card-image {
  height: 100%;
  width: 100%;
  position: absolute;
  right: 0;
  bottom: 0;
  object-fit: contain;
  z-index: 0;
  transform: scale(0.8);
  filter: brightness(1) drop-shadow(0 0 5px var(--CL-neutral-black));
}

.card-actions,
.card-data {
  position: relative;
  z-index: 100;
  text-align: right;
  color: var(--CL-neutral-white);
}

.card-button {
  font-size: 1.5rem;
  padding: 0;
  color: var(--CL-neutral-white);
}

.card-data > * {
  margin: 0;
}

.card-title {
  font-weight: bold;
  font-size: 1.35rem;
}

.card-count {
  font-size: 1rem;
  text-align: right;
}

@media (min-width: 768px) {
  .card {
    height: 20em;
    padding: 2em;
    transition: 0.15s;
  }

  .card-title {
    font-size: 3rem;
  }

  .card-count {
    font-size: 2rem;
  }

  .card-image {
    left: 2em;
    width: max-content;
    transition: 0.15s;
  }

  .card:hover {
    background-color: var(--CL-neutral-black);
  }

  .card:hover .card-image {
    transform: scale(1.8) translateY(-5em) translateX(65%);
  }
}
