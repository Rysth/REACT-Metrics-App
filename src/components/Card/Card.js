import React from 'react';
import PropTypes from 'prop-types';
import './Card.css';

function Card(
  /* prettier-ignore */
  {
    className,
    count,
    imageSource,
  },
) {
  return (
    <div className="card bg-primary-blue text-white">
      <img className="card-image" src={imageSource} alt="" />
      <div className="card-actions">
        <button type="button" className="card-button">
          <i className="fa-solid fa-circle-arrow-right" />
        </button>
      </div>
      <div className="card-data">
        <h3 className="card-title">{className}</h3>
        <p className="cart-count">{count}</p>
      </div>
    </div>
  );
}

Card.propTypes = {
  className: PropTypes.string.isRequired,
  count: PropTypes.number.isRequired,
  imageSource: PropTypes.string.isRequired,
};

export default Card;
