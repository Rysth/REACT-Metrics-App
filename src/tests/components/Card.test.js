import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import Card from '../../components/Card/Card';

test('Card component snapshot', () => {
  const { container } = render(
    <Router>
      <Card
        slug="example-slug"
        title="Example Title"
        count={42}
        imageSource="example-image.jpg"
      />
    </Router>,
  );
  expect(container).toMatchInlineSnapshot(`
<div>
  <div
    style="opacity: 0;"
  >
    <a
      class="card bg-primary-blue"
      href="/class/example-slug"
    >
      <img
        alt=""
        class="card-image"
        src="example-image.jpg"
      />
      <div
        class="card-actions"
      >
        <button
          class="card-button"
          type="button"
        >
          <i
            class="fa-solid fa-circle-arrow-right"
          />
        </button>
      </div>
      <div
        class="card-data"
      >
        <h3
          class="card-title"
        >
          Example Title
        </h3>
        <p
          class="card-count"
        >
          42
        </p>
      </div>
    </a>
  </div>
</div>
`);
});

test('Card component is in the document', () => {
  const { container } = render(
    <Router>
      <Card
        slug="example-slug"
        title="Example Title"
        count={42}
        imageSource="example-image.jpg"
      />
    </Router>,
  );

  expect(container.querySelector('.card')).toBeInTheDocument();
});

test('Props should not be null', () => {
  const { getByText } = render(
    <Router>
      <Card
        slug="example-slug"
        title="Example Title"
        count={42}
        imageSource="example-image.jpg"
      />
    </Router>,
  );

  expect(getByText('Example Title')).not.toBeNull();
  expect(getByText('42')).not.toBeNull();
});
