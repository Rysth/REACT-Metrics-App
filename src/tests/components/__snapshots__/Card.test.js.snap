// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component snapshot 1`] = `
Array [
  <div
    style={
      Object {
        "opacity": 0,
      }
    }
  >
    <a
      className="card bg-primary-blue"
      href="/class/example-slug"
      onClick={[Function]}
    >
      <img
        alt=""
        className="card-image"
        src="example-image.jpg"
      />
      <div
        className="card-actions"
      >
        <button
          className="card-button"
          type="button"
        >
          <i
            className="fa-solid fa-circle-arrow-right"
          />
        </button>
      </div>
      <div
        className="card-data"
      >
        <h3
          className="card-title"
        >
          Example Title
        </h3>
        <p
          className="card-count"
        >
          42
        </p>
      </div>
    </a>
  </div>,
  ",",
]
`;
